FreeRTOS/FreeRTOS
# Port shadow demo from product 4

Description
-----------

This PR port shadow demo from (another PR for amazon freertos](https://github.com/aws/amazon-freertos/pull/2537).

Test Steps
-----------

setup required configuration in demo_config.h, then this demo can be built and run successfully.

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
